File: adapter/java/concrete/Mp4Player.java
package adapter.java.concrete;

import adapter.java.legacy.AdvancedMediaPlayer;

public class Mp4Player implements AdvancedMediaPlayer{
    // Implementacion concreta de Adaptee
    @Override
    public void playVlc(String fileName) {
        // No hace nada
    }

    @Override
    public void playMp4(String fileName) {
        System.out.println("Reproduciendo archivo mp4. Nombre: " + fileName);
    }
}

File: adapter/php/concrete/Mp4Player.php
<?php
namespace Concrete;

require_once 'legacy/AdvancedMediaPlayer.php';
use Legacy\AdvancedMediaPlayer;

//Implementacion concreta de Adaptee
class Mp4Player implements AdvancedMediaPlayer {
    public function playVlc($fileName) {
        // no hace nada
    }
    public function playMp4($fileName) {
        echo "Reproduciendo archivo mp4. Nombre: " . $fileName . "\n";
    }
}